### PR TITLE
Update Anypoint MQ Broker API working mechanism

### PR DESCRIPTION
**Pull request by macrocks**
Anypoint broker api doesn't gurantee that it will pull 10messages in a batch size of 10 as it depends on underlying systems who holds messages and their availability.

It took lot of time for me to identify this and wasted my efforts to a put solution design in place

### CODE REVIEW

**Comment by hannanelson** [comment]
Hi @schinthakindims - can you verify that this update is correct? Thanks!

**Comment by GouthamiKondakindi** [comment]
Hi @macrocks, this issue is already explained in detail in this FAQ: https://docs.mulesoft.com/mq/mq-faq#batchsize. 

@hannanelson do you think we should update the FAQ mentioned in this PR and link to mq-faq#batchsize for additional details or would that be repetitive?

**Review by schinthakindims — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/mq-faq.adoc [hunk 1]
```diff
@@ -201,7 +201,7 @@ Using the Anypoint MQ Broker API, you can batch-upload and batch-retrieve messag
 
 You can upload or retrieve a maximum of 10 messages (default) in a single call.
 
-For batch-retrieve operations, use the `batchSize` query parameter to specify the number of messages to retrieve.
+For batch-retrieve operations, use the `batchSize` query parameter to specify the number of messages to retrieve. Anypoint Broker API can't guarantee it will retrieve 10 messages in a single call as it depends on underlying system's polling mechanism and availability.
```
Comment (hannanelson) [inline modules/ROOT/pages/mq-faq.adoc]:
```suggestion
For batch-retrieve operations, use the `batchSize` query parameter to specify the number of messages to retrieve.
Anypoint Broker API doesn't guarantee the number of messages retrieved in a single call.
For more information, see <<batchsize,Why do I receive fewer messages than my batch size when retrieving messages with the Anypoint MQ REST API?>>.
```

**Comment by hannanelson** [comment]
Good point, @GouthamiKondakindi. I removed the redundant explanation and added a link. Please rereview.

File: modules/ROOT/pages/mq-faq.adoc [hunk 2]
```diff
@@ -198,13 +198,14 @@ Retries are arbitrary. The `maxRedelivery` parameter indicates how many times th
...
+For information, see 
+https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/anypoint-mq-broker/minor/1.0/console/summary/[Anypoint MQ Broker API].
 
 You can upload or retrieve a maximum of 10 messages (default) in a single call.
 
 For batch-retrieve operations, use the `batchSize` query parameter to specify the number of messages to retrieve.
-
-For more information, see 
-https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/anypoint-mq-broker/minor/1.0/console/summary/[Anypoint MQ Broker API].
+Anypoint Broker API doesn't guarantee the number of messages retrieved in a single call.
```
Comment (GouthamiKondakindi) [inline modules/ROOT/pages/mq-faq.adoc]: "Anypoint MQ Broker API" instead of "Anypoint Broker API"?

Comment (hannanelson) [inline modules/ROOT/pages/mq-faq.adoc]:
```suggestion
Anypoint MQ Broker API doesn't guarantee the number of messages retrieved in a single call.
```

Comment (hannanelson) [reply modules/ROOT/pages/mq-faq.adoc]: Oy! Good catch. Thank you! @GouthamiKondakindi

**Review by GouthamiKondakindi — APPROVED** [review]
LGTM

**Review by kbostrom — APPROVED** [review]
PR

**Pull request merged**